src/main/scala/amba/axi4/UserYanker.scala

```diff
@@ -8,6 +8,14 @@ import freechips.rocketchip.config.Parameters
 import freechips.rocketchip.diplomacy._
 import freechips.rocketchip.util._
 
+/** This adapter prunes all user bit fields of the echo type from request messages,
+  * storing them in queues and echoing them back when matching response messages are recevied.
+  *
+  * It also optionally rate limits the number of transactions that can be in flight simultaneously
+  * per FIFO domain / AXID.
+  *
+  * @param capMaxFlight is an optional maximum number of transactions that can be in flight per AXID.
```

Comment:
AXI Id?

Reply:
hmm "AXID" is their term for `A[W|R]ID` which is the actual signal field name but maybe i will change it to be the latter for added clarity

Reply:
ah, thats fine